UKHomeOffice/sgmr-service
# Cg 121 remove people from voyage

TO TEST

- go to _reports > start now_
- fill out the form
- - the only required fields at this stage are vessel name & people
- on the _'Check all the information...'?_ page
- - check your people are there
- - click _'Change'_ for the people section
> you should see the people page of the form

----

**Can you remove a person**
- remove a person by unchecking their checkbox
- click add/save button
- continue through form to the check information page
> your person should have been removed

----

**Can you add a person**
- click _'Change'_ for the people section
- add a person by checking their checkbox
- continue through form to the check information page
> your person should have been added

----

**Can you add and remove people in one action**
- click _'Change'_ for the people section
- add a person by checking their checkbox
- remove a person by unchecking their checkbox
- continue through form to the check information page
> your checked person should have been added  & your unchecked person removed

----

**Can you edit people in a voyage**
- Return to _/reports_
- click on the vessel name of your voyage
- repeat the tests above
- - remove a person
- - add a person
- - add/remove in one action

----

THINGS NOT COVERED IN THIS PR

- required fields are tested on each page before PATCH
- - cg-226/229: all fields required have data in them
- - cg-227: People/Manifest page ensures person has all required fields
- - cg-228: dates are validated
- - cg-196: if sign-out occurs during form it returns you to the correct page

- ensure source is being stored correctly
- - cg-196: ensure /people doesn't overwrite /voyage if editing/adding person (this is connected to how /source is stored which is part of cg-196)

BUGS:

- sometimes dates add an extra month on in the UI view : I think it's to do with formatting so will address as part of next PR which validates dates
- cg-198: on first login you are not always taken to /reports
- cg-216: clear redux store on logout
